Sources/Sentry/SentryHttpTransport.m

```diff
@@ -76,16 +80,36 @@ - (id)initWithOptions:(SentryOptions *)options
         [self.fileManager setDelegate:self];
 
         [self sendAllCachedEnvelopes];
+
+#if !TARGET_OS_WATCH
+        [SentryReachability
```

Comment:
`h`: This won't work if you have multiple clients or multiple hubs. The way how `SentryReachability` works right now is that it only stores one callback, which is the latest one. We could keep track of a list of callbacks as we do like for example here https://github.com/getsentry/sentry-cocoa/blob/372ffbef48535f7179f7120c431887b8aa98018a/Sources/Sentry/SentrySwizzleWrapper.m#L9-L10